test/test_cuda.py

```diff
@@ -4407,6 +4408,35 @@ class TestNamedTupleInput_1(NamedTuple):
             cat = torch.cat((outputs[0][i].to('cpu'), outputs[1][i].to('cpu')))
             self.assertTrue(torch.equal(x, cat))
 
+    def test_memory_snapshot(self):
+        try:
+            torch.cuda.memory.empty_cache()
+            torch.cuda.memory._record_memory_history(True)
+            x = torch.rand(311, 411).cuda()
+            ss = torch.cuda.memory._snapshot()
+
+            tensors = [torch.rand(128, device='cuda') for _ in range(1000)]
+            while tensors:
+                del tensors[randint(0, len(tensors) - 1)]
+            torch.rand(128 * 5, device='cuda')
```

Comment:
This is a really big Tensor for testing? Can we reduce all the sizes here or they are tuned to match the allocator block sizes? If so, can you leave a comment here to specific that?

Reply:
It's the latter 128*sizeof(float) == 512, which is the block size these allocations would get rounded to otherwise. I'll comment.